### PR TITLE
Version 1.0.0 of MMTF changes

### PR DESCRIPTION
**Pull request by abradle**
- Fixed issues with reading a few MMTF files (related to hetatm vs amino)
- Fixed a bug in generation of disulphide bonds - including a test case.
- Fixed a null pointer exception from getAtomFromRecord
- Fixed handling of entities with no chains
- Upgrade to v1.0.0
- Fixed the formatting and added some javadocs

### CODE REVIEW

File: biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmtf/MmtfStructureReader.java [hunk 1]
```diff
@@ -411,7 +411,13 @@ private int getGroupTypIndicator(String currentGroupType) {
 		}
 		if(currentGroupType.toUpperCase().contains("D-GAMMA-PEPTIDE")){
 			return 0;
-		}		
+		}
+		if(currentGroupType.toUpperCase().contains("D-BETA-PEPTIDE")){
```
Comment (andreasprlic) [inline biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmtf/MmtfStructureReader.java]: Would be better to use the ResidueType enum here.

**Comment by andreasprlic** [comment]
I think it would be better to replace getGroupTypIndicator(String) with the ResidueType.getResidueTypeFromString. Then it is possible to use the built-in ResidueType and PolymerType classes to identify if a group is an AminoAcid, Nucleotide, or Hetatom. In addition this ensures consistency with the CCD.

**Comment by abradle** [comment]
Thanks Andreas! Good point - fixed.

**Comment by andreasprlic** [comment]
Great thanks!

**Pull request merged**